rs/execution_environment/src/execution_environment/tests/canister_task.rs

```diff
@@ -819,3 +820,253 @@ fn global_timer_produces_transient_error_on_out_of_cycles() {
 
     assert_eq!(RejectCode::SysTransient, err.code().into());
 }
+
+#[test]
```

Comment:
The last test that I think is missing is one where there's a DTS execution to ensure that this still gets completed before the hook is triggered.

Reply:
The test is added.